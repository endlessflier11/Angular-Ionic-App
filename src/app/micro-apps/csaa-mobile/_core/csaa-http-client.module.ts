import { NgModule } from '@angular/core';
import { AngularHttpAdapter } from './services/http/adapters/angular-http.adapter';
import { HttpClient, HttpClientModule } from '@angular/common/http';
import { IonicHttpAdapter } from './services/http/adapters/ionic-http.adapter';
import { HTTP } from '@ionic-native/http/ngx';
import { UNI_HTTP_ADAPTER } from './services/http/uni-http.model';

@NgModule({
  imports: [HttpClientModule],
  providers: [
    HTTP,
    { provide: UNI_HTTP_ADAPTER, multi: true, useClass: AngularHttpAdapter, deps: [HttpClient] },
    {
      provide: UNI_HTTP_ADAPTER,
      multi: true,
      useClass: IonicHttpAdapter,
      deps: [HTTP],
    },
  ],
  exports: [HttpClientModule],
})
export class CsaaHttpClientModule {}
