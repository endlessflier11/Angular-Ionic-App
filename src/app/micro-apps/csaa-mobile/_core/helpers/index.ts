export * from './router.helpers';
export * from './misc.helpers';
export * from './async.helper';
export * from './error.helper';
export * from './string.helper';
