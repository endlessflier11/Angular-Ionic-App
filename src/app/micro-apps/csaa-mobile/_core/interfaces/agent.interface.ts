export interface Agent {
  name: string;
  phone: string;
  email: string;
  isService: boolean;
}
