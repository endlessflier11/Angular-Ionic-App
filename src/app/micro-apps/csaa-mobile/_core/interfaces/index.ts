export * from './auth.interface';
export * from './router';
export * from './http.interface';
export * from './vehicle.interface';
export * from './ionic-lifecycle.interface';
export * from './analytics.interface';
export * from './policy.interface';
export * from './payment.interface';
export * from './config.interface';
export * from './ngxs-store.interface';
