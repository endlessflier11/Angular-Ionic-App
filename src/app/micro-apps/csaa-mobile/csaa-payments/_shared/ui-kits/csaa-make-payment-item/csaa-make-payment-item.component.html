<div class="csaa-make-payment-item">
  <ion-grid>
    <ion-row class="item" *ngIf="payment?.isPaymentDue">
      <ion-col>
        <span class="payment-due" data-test="payment-due" [class.overdue]="payment?.isPastDue">
          <ng-container *ngIf="payment?.isPastDue">Past</ng-container> Due
          {{ payment?.dueDate | date: 'longDate':timezone }}
        </span>
      </ion-col>
      <ion-col size="auto">
        <span class="payment-due" data-test="payment-amount" [class.overdue]="payment?.isPastDue">{{
          payment?.amount | currency: 'USD':'symbol':'1.2-2'
        }}</span>
      </ion-col>
    </ion-row>
    <ion-row class="item" *ngIf="!payment?.isPaymentDue">
      <ion-col>
        <span class="payment-due" data-test="payment-due"> No Payment Due </span>
      </ion-col>
    </ion-row>
    <ion-row class="item">
      <ion-col>
        <div class="policy-meta-content">
          <span class="subtitle policy-number" data-test="policy-number">
            {{ payment?.policyNumber }}
          </span>
          <span class="subtitle">
            <csaa-policy-subtitles [policySubtitle]="payment?.subtitle"></csaa-policy-subtitles>
          </span>
        </div>
      </ion-col>
    </ion-row>
  </ion-grid>

  <ion-list [class.overdue]="isOverdue()">
    <ion-item
      detail
      class="ion-no-padding"
      data-test="button-pay-now"
      (click)="payNowClick.emit(payment)"
    >
      <ion-label>Pay now</ion-label>
    </ion-item>
    <ion-item
      detail
      class="csaa-payment-autopay-item ion-no-padding"
      *ngIf="hasRegId()"
      data-test="button-autopay"
      (click)="autoPayClick.emit(payment)"
    >
      <ion-label>
        <label>AutoPay</label>
        <small *ngIf="payment?.autoPay">
          {{ getAutopayInfo() }}
        </small>
      </ion-label>
      <span slot="end"> {{ payment?.autoPay ? 'On' : 'Off' }}</span>
    </ion-item>
  </ion-list>
</div>
