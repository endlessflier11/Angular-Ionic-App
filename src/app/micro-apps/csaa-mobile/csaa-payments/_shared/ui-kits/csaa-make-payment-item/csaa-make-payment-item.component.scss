@import '../../../../_scss/variables';
.csaa-make-payment-item {
  padding-bottom: 10px;
  padding-top: 20px;
}

span {
  &.payment-due {
    font-size: $csaa-subheader-font-size;
    font-weight: $csaa-bolder-font-weight !important;

    &.overdue {
      color: $csaa-emergency-color !important;
    }
  }

  &.policy-number {
    color: $csaa-auto-id-card-bg-primary;
    font-weight: bold;
  }
}

ion-list {
  margin-top: 8px;
  ion-item {
    font-size: $csaa-subheader-font-size !important;
    --inner-border-width: 0 0 0 0;

    &.csaa-payment-autopay-item {
      ion-label {
        display: flex !important;
        flex-flow: column nowrap;
        small {
          margin-top: 3px;
          color: $csaa-lighter-color;
        }
      }

      span[slot='end'] {
        color: $csaa-lighter-color;
      }
    }
  }
}

.policy-meta-content {
  margin-top: 5px;
}
