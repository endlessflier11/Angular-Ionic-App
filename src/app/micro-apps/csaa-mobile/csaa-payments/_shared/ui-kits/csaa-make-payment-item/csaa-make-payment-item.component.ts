import { Component, EventEmitter, Input, Output } from '@angular/core';
import {
  AutoPayEnrollmentResponse,
  CustomerSearchResponse,
  PaymentAccount,
  UpcomingPayment,
} from '../../../../_core/interfaces';
import { PaymentHelper } from '../../../../_core/shared/payment.helper';

@Component({
  selector: 'csaa-make-payment-item',
  templateUrl: './csaa-make-payment-item.component.html',
  styleUrls: ['./csaa-make-payment-item.component.scss'],
})
export class CsaaMakePaymentItemComponent {
  @Input() paymentSucceeded = false;
  @Input() timezone: string = undefined;
  @Input() payment: UpcomingPayment;
  @Input() customerSearch: CustomerSearchResponse;
  @Input() loading = false;
  @Input() paymentMethod: PaymentAccount;
  @Input() otherAmount: number;
  @Input() autopayEnrollment: AutoPayEnrollmentResponse;

  @Output() payNowClick = new EventEmitter<UpcomingPayment>();
  @Output() autoPayClick = new EventEmitter<UpcomingPayment>();

  hasRegId() {
    if (
      this.customerSearch &&
      this.customerSearch.registrations &&
      this.customerSearch.registrations.length > 0
    ) {
      if (this.customerSearch.registrations.filter((reg) => reg.registrationId).length > 0) {
        return true;
      }
    }
    return false;
  }

  isOverdue(): boolean {
    if (!this.payment) {
      return false;
    }

    return this.payment.isPastDue;
  }

  getAutopayInfo() {
    return PaymentHelper.getAutopayEnrollmentLabel(this.autopayEnrollment);
  }
}
