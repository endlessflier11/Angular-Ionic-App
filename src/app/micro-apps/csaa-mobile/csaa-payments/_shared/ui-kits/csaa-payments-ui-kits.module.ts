import { NgModule } from '@angular/core';
import { IonicModule } from '@ionic/angular';
import { CommonModule } from '@angular/common';
import { CsaaPaymentHistoryCardComponent } from './csaa-payment-history-card/csaa-payment-history-card.component';
import { UiKitsModule } from '../../../_core/ui-kits/ui-kits.module';
import { CsaaPaymentHistorySelectCardComponent } from './csaa-payment-history-select-card/csaa-payment-history-select-card.component';
import { CsaaPaymentSucceededCardComponent } from './csaa-payment-succeeded-card/csaa-payment-succeeded-card.component';
import { CsaaMakePaymentCardComponent } from './csaa-make-payment-card/csaa-make-payment-card.component';
import { CsaaPayAllPoliciesCardComponent } from './csaa-pay-all-policies-card/csaa-pay-all-policies-card.component';
import { PaymentTermsConditionsComponent } from './payment-terms-conditions/payment-terms-conditions.component';
import { AutopayTermsConditionsComponent } from './autopay-terms-conditions/autopay-terms-conditions.component';
import { CsaaBillsCardComponent } from './csaa-bills-card/csaa-bills-card.component';
import { AutopayDiscountCardComponent } from './autopay-discount-card/autopay-discount-card.component';
import { CsaaMakePaymentItemComponent } from './csaa-make-payment-item/csaa-make-payment-item.component';
import { IconPaymentMethodComponent } from './icon-payment-method/icon-payment-method.component';
import { CardSelectAmountComponent } from './card-select-amount/card-select-amount.component';
import { BrMaskerModule } from 'br-mask';
import { ReactiveFormsModule } from '@angular/forms';
import { CardGroupPaymentMethodComponent } from './card-group-payment-method/card-group-payment-method.component';

const SHARED_COMPONENTS = [
  CsaaPaymentHistoryCardComponent,
  CsaaPaymentHistorySelectCardComponent,
  CsaaMakePaymentCardComponent,
  CsaaPaymentSucceededCardComponent,
  CsaaPayAllPoliciesCardComponent,
  PaymentTermsConditionsComponent,
  AutopayTermsConditionsComponent,
  CsaaBillsCardComponent,
  AutopayDiscountCardComponent,
  CsaaMakePaymentItemComponent,
];

@NgModule({
  declarations: [
    ...SHARED_COMPONENTS,
    IconPaymentMethodComponent,
    CardSelectAmountComponent,
    CardGroupPaymentMethodComponent,
  ],
  imports: [
    CommonModule,
    IonicModule,
    UiKitsModule,
    IonicModule,
    BrMaskerModule,
    ReactiveFormsModule,
  ],
  exports: [
    ...SHARED_COMPONENTS,
    IconPaymentMethodComponent,
    CardSelectAmountComponent,
    CardGroupPaymentMethodComponent,
  ],
})
export class CsaaPaymentsUiKitsModule {}
