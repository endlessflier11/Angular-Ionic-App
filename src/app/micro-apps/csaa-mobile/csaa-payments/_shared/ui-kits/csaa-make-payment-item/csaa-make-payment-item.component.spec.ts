import { ComponentFixture, TestBed } from '@angular/core/testing';
import { CsaaMakePaymentItemComponent } from './csaa-make-payment-item.component';
import { CommonModule } from '@angular/common';
import { IonicModule } from '@ionic/angular';
import { UiKitsModule } from '../../../../_core/ui-kits/ui-kits.module';
import { PaymentAccountType, PaymentType, PolicyType } from '../../../../_core/interfaces';
import DateHelper from '../../../../_core/shared/date.helper';
import { By, clickEvent } from '@app/testing';
import { MAKE_PAYMENT_METHOD } from '../../../../../../../testing/fixtures/make-payments-page/make-payments-page.fixture';
import { EMPLOYEE_SEARCH } from '../../../../../../../testing/fixtures/make-payments-page/employee-search.fixture';

function createPaymentMock() {
  return {
    vehicles: ['Toyota Prius'],
    subtitle: 'test',
    autoPay: true,
    minimumAmount: 20,
    remainingPremium: 10000,
    otherAmount: 1000,
    type: PaymentType.other,
    nextDueDate: DateHelper.toDate('2040-08-19'),
    nextAmount: 30,
    allPolicies: false,
    policyNumber: 'AZSS294857564',
    policyType: PolicyType.Auto,
    dueDate: DateHelper.toDate('2040-08-19'),
    isPastDue: false,
    isPaymentDue: false,
    amount: 148.1,
    policyRiskState: 'test',
    autopayEnrollment: {
      enrollmentRecord: {
        receiptNumber: '755736',
        action: 'ENROLL',
        authenticationChannel: null,
        sourceApplicationUsed: 'EZPAY',
        lastUpdatedBy: 'mobile_app_user',
        lastUpdatedOn: '2021-10-22T07:26:05.757-07:00',
        lastUpdatedWorkstationIP: '127.0.0.1',
        billingSystemUpdateStatus: 'POST',
        enrollmentEffectiveDate: '2021-10-22',
        paymentItem: {
          paymentMethod: PaymentAccountType.CARD,
          type: 'INST',
          paymentAccountToken: '0AC5D0B0-5383-BBB2-6EF6-0E7C35B81E7D',
          card: {
            number: '5454',
            type: 'MASTR',
            printedName: 'TEST USER',
            expirationDate: '2025-12-01',
            isCardPresent: false,
            zipCode: null,
          },
          account: null,
        },
      },
      policyNumber: 'AZAC910055047',
      dataSource: 'PAS',
      policyType: 'PA',
      autoPay: true,
    },
  };
}

describe('CsaaMakePaymentItemComponent', () => {
  const DATE_NOW = Date.parse('2021-01-01');
  let component: CsaaMakePaymentItemComponent;
  let fixture: ComponentFixture<CsaaMakePaymentItemComponent>;

  beforeEach(async () => {
    jest.spyOn(Date, 'now').mockReturnValue(DATE_NOW);
    await TestBed.configureTestingModule({
      imports: [CommonModule, IonicModule, UiKitsModule],
      declarations: [CsaaMakePaymentItemComponent],
    }).compileComponents();
  });
  afterEach(() => {
    jest.spyOn(Date, 'now').mockRestore();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(CsaaMakePaymentItemComponent);
    component = fixture.componentInstance;
    component.paymentMethod = MAKE_PAYMENT_METHOD;
    component.customerSearch = EMPLOYEE_SEARCH;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshots when payment is due', () => {
    component.payment = createPaymentMock();
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
  it('should match snapshots when payment is overdue', () => {
    component.payment = {
      ...createPaymentMock(),
      vehicles: ['Lexus NX', 'BMW X1'],
      isPastDue: true,
      isPaymentDue: true,
    };

    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should match snapshots when autopay is off', () => {
    component.payment = { ...createPaymentMock(), autoPay: false, autopayEnrollment: null };

    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should have an Auto Pay button', () => {
    component.payment = createPaymentMock();
    fixture.detectChanges();
    const autoPayButton = fixture.debugElement.query(By.css('[data-test=button-autopay]'));
    expect(autoPayButton).toBeTruthy();
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should emit autoPayClick event when AutoPay button clicked', () => {
    component.payment = { ...createPaymentMock(), isPastDue: false, isPaymentDue: true };
    component.autoPayClick = { emit: jest.fn() } as any;
    fixture.detectChanges();

    const autopayButton = fixture.debugElement.query(By.css('[data-test=button-autopay]'));
    expect(autopayButton).toBeTruthy();

    autopayButton.triggerEventHandler('click', clickEvent);

    expect(component.autoPayClick.emit).toHaveBeenCalled();

    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should have a Pay Now button', () => {
    component.payment = { ...createPaymentMock(), isPastDue: false, isPaymentDue: true };
    fixture.detectChanges();

    const makeAPaymentButton = fixture.debugElement.query(By.css('[data-test=button-pay-now]'));
    expect(makeAPaymentButton).toBeTruthy();
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should emit payNowClick event when Pay Now button clicked', () => {
    component.payment = { ...createPaymentMock(), isPastDue: false, isPaymentDue: true };
    component.payNowClick = { emit: jest.fn() } as any;
    fixture.detectChanges();

    const payNowButton = fixture.debugElement.query(By.css('[data-test=button-pay-now]'));
    expect(payNowButton).toBeTruthy();

    payNowButton.triggerEventHandler('click', clickEvent);

    expect(component.payNowClick.emit).toHaveBeenCalled();

    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
