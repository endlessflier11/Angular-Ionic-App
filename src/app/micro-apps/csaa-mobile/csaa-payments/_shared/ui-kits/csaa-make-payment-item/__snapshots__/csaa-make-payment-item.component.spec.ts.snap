// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CsaaMakePaymentItemComponent should emit autoPayClick event when AutoPay button clicked 1`] = `
<csaa-make-payment-item
  __ngContext__={[Function LRootView]}
  autoPayClick={[Function Object]}
  customerSearch={[Function Object]}
  loading="false"
  payNowClick={[Function EventEmitter_]}
  payment={[Function Object]}
  paymentMethod={[Function Object]}
  paymentSucceeded="false"
  timezone="undefined"
>
  <div
    class="csaa-make-payment-item"
  >
    <ion-grid>
      <ion-row
        class="item"
      >
        <ion-col>
          <span
            class="payment-due"
            data-test="payment-due"
          >
             Due August 19, 2040 
          </span>
        </ion-col>
        <ion-col
          size="auto"
        >
          <span
            class="payment-due"
            data-test="payment-amount"
          >
            $148.10
          </span>
        </ion-col>
      </ion-row>
      <ion-row
        class="item"
      >
        <ion-col>
          <div
            class="policy-meta-content"
          >
            <span
              class="subtitle policy-number"
              data-test="policy-number"
            >
               AZSS294857564 
            </span>
            <span
              class="subtitle"
            >
              <csaa-policy-subtitles>
                <span
                  class="subtitle"
                >
                  test
                </span>
              </csaa-policy-subtitles>
            </span>
          </div>
        </ion-col>
      </ion-row>
    </ion-grid>
    <ion-list>
      <ion-item
        class="ion-no-padding"
        data-test="button-pay-now"
        detail=""
      >
        <ion-label>
          Pay now
        </ion-label>
      </ion-item>
      <ion-item
        class="csaa-payment-autopay-item ion-no-padding"
        data-test="button-autopay"
        detail=""
      >
        <ion-label>
          <label>
            AutoPay
          </label>
          <small>
          </small>
        </ion-label>
        <span
          slot="end"
        >
           On
        </span>
      </ion-item>
    </ion-list>
  </div>
</csaa-make-payment-item>
`;

exports[`CsaaMakePaymentItemComponent should emit payNowClick event when Pay Now button clicked 1`] = `
<csaa-make-payment-item
  __ngContext__={[Function LRootView]}
  autoPayClick={[Function EventEmitter_]}
  customerSearch={[Function Object]}
  loading="false"
  payNowClick={[Function Object]}
  payment={[Function Object]}
  paymentMethod={[Function Object]}
  paymentSucceeded="false"
  timezone="undefined"
>
  <div
    class="csaa-make-payment-item"
  >
    <ion-grid>
      <ion-row
        class="item"
      >
        <ion-col>
          <span
            class="payment-due"
            data-test="payment-due"
          >
             Due August 19, 2040 
          </span>
        </ion-col>
        <ion-col
          size="auto"
        >
          <span
            class="payment-due"
            data-test="payment-amount"
          >
            $148.10
          </span>
        </ion-col>
      </ion-row>
      <ion-row
        class="item"
      >
        <ion-col>
          <div
            class="policy-meta-content"
          >
            <span
              class="subtitle policy-number"
              data-test="policy-number"
            >
               AZSS294857564 
            </span>
            <span
              class="subtitle"
            >
              <csaa-policy-subtitles>
                <span
                  class="subtitle"
                >
                  test
                </span>
              </csaa-policy-subtitles>
            </span>
          </div>
        </ion-col>
      </ion-row>
    </ion-grid>
    <ion-list>
      <ion-item
        class="ion-no-padding"
        data-test="button-pay-now"
        detail=""
      >
        <ion-label>
          Pay now
        </ion-label>
      </ion-item>
      <ion-item
        class="csaa-payment-autopay-item ion-no-padding"
        data-test="button-autopay"
        detail=""
      >
        <ion-label>
          <label>
            AutoPay
          </label>
          <small>
          </small>
        </ion-label>
        <span
          slot="end"
        >
           On
        </span>
      </ion-item>
    </ion-list>
  </div>
</csaa-make-payment-item>
`;

exports[`CsaaMakePaymentItemComponent should have a Pay Now button 1`] = `
<csaa-make-payment-item
  __ngContext__={[Function LRootView]}
  autoPayClick={[Function EventEmitter_]}
  customerSearch={[Function Object]}
  loading="false"
  payNowClick={[Function EventEmitter_]}
  payment={[Function Object]}
  paymentMethod={[Function Object]}
  paymentSucceeded="false"
  timezone="undefined"
>
  <div
    class="csaa-make-payment-item"
  >
    <ion-grid>
      <ion-row
        class="item"
      >
        <ion-col>
          <span
            class="payment-due"
            data-test="payment-due"
          >
             Due August 19, 2040 
          </span>
        </ion-col>
        <ion-col
          size="auto"
        >
          <span
            class="payment-due"
            data-test="payment-amount"
          >
            $148.10
          </span>
        </ion-col>
      </ion-row>
      <ion-row
        class="item"
      >
        <ion-col>
          <div
            class="policy-meta-content"
          >
            <span
              class="subtitle policy-number"
              data-test="policy-number"
            >
               AZSS294857564 
            </span>
            <span
              class="subtitle"
            >
              <csaa-policy-subtitles>
                <span
                  class="subtitle"
                >
                  test
                </span>
              </csaa-policy-subtitles>
            </span>
          </div>
        </ion-col>
      </ion-row>
    </ion-grid>
    <ion-list>
      <ion-item
        class="ion-no-padding"
        data-test="button-pay-now"
        detail=""
      >
        <ion-label>
          Pay now
        </ion-label>
      </ion-item>
      <ion-item
        class="csaa-payment-autopay-item ion-no-padding"
        data-test="button-autopay"
        detail=""
      >
        <ion-label>
          <label>
            AutoPay
          </label>
          <small>
          </small>
        </ion-label>
        <span
          slot="end"
        >
           On
        </span>
      </ion-item>
    </ion-list>
  </div>
</csaa-make-payment-item>
`;

exports[`CsaaMakePaymentItemComponent should have an Auto Pay button 1`] = `
<csaa-make-payment-item
  __ngContext__={[Function LRootView]}
  autoPayClick={[Function EventEmitter_]}
  customerSearch={[Function Object]}
  loading="false"
  payNowClick={[Function EventEmitter_]}
  payment={[Function Object]}
  paymentMethod={[Function Object]}
  paymentSucceeded="false"
  timezone="undefined"
>
  <div
    class="csaa-make-payment-item"
  >
    <ion-grid>
      <ion-row
        class="item"
      >
        <ion-col>
          <span
            class="payment-due"
            data-test="payment-due"
          >
             No Payment Due 
          </span>
        </ion-col>
      </ion-row>
      <ion-row
        class="item"
      >
        <ion-col>
          <div
            class="policy-meta-content"
          >
            <span
              class="subtitle policy-number"
              data-test="policy-number"
            >
               AZSS294857564 
            </span>
            <span
              class="subtitle"
            >
              <csaa-policy-subtitles>
                <span
                  class="subtitle"
                >
                  test
                </span>
              </csaa-policy-subtitles>
            </span>
          </div>
        </ion-col>
      </ion-row>
    </ion-grid>
    <ion-list>
      <ion-item
        class="ion-no-padding"
        data-test="button-pay-now"
        detail=""
      >
        <ion-label>
          Pay now
        </ion-label>
      </ion-item>
      <ion-item
        class="csaa-payment-autopay-item ion-no-padding"
        data-test="button-autopay"
        detail=""
      >
        <ion-label>
          <label>
            AutoPay
          </label>
          <small>
          </small>
        </ion-label>
        <span
          slot="end"
        >
           On
        </span>
      </ion-item>
    </ion-list>
  </div>
</csaa-make-payment-item>
`;

exports[`CsaaMakePaymentItemComponent should match snapshots when autopay is off 1`] = `
<csaa-make-payment-item
  __ngContext__={[Function LRootView]}
  autoPayClick={[Function EventEmitter_]}
  customerSearch={[Function Object]}
  loading="false"
  payNowClick={[Function EventEmitter_]}
  payment={[Function Object]}
  paymentMethod={[Function Object]}
  paymentSucceeded="false"
  timezone="undefined"
>
  <div
    class="csaa-make-payment-item"
  >
    <ion-grid>
      <ion-row
        class="item"
      >
        <ion-col>
          <span
            class="payment-due"
            data-test="payment-due"
          >
             No Payment Due 
          </span>
        </ion-col>
      </ion-row>
      <ion-row
        class="item"
      >
        <ion-col>
          <div
            class="policy-meta-content"
          >
            <span
              class="subtitle policy-number"
              data-test="policy-number"
            >
               AZSS294857564 
            </span>
            <span
              class="subtitle"
            >
              <csaa-policy-subtitles>
                <span
                  class="subtitle"
                >
                  test
                </span>
              </csaa-policy-subtitles>
            </span>
          </div>
        </ion-col>
      </ion-row>
    </ion-grid>
    <ion-list>
      <ion-item
        class="ion-no-padding"
        data-test="button-pay-now"
        detail=""
      >
        <ion-label>
          Pay now
        </ion-label>
      </ion-item>
      <ion-item
        class="csaa-payment-autopay-item ion-no-padding"
        data-test="button-autopay"
        detail=""
      >
        <ion-label>
          <label>
            AutoPay
          </label>
        </ion-label>
        <span
          slot="end"
        >
           Off
        </span>
      </ion-item>
    </ion-list>
  </div>
</csaa-make-payment-item>
`;

exports[`CsaaMakePaymentItemComponent should match snapshots when payment is due 1`] = `
<csaa-make-payment-item
  __ngContext__={[Function LRootView]}
  autoPayClick={[Function EventEmitter_]}
  customerSearch={[Function Object]}
  loading="false"
  payNowClick={[Function EventEmitter_]}
  payment={[Function Object]}
  paymentMethod={[Function Object]}
  paymentSucceeded="false"
  timezone="undefined"
>
  <div
    class="csaa-make-payment-item"
  >
    <ion-grid>
      <ion-row
        class="item"
      >
        <ion-col>
          <span
            class="payment-due"
            data-test="payment-due"
          >
             No Payment Due 
          </span>
        </ion-col>
      </ion-row>
      <ion-row
        class="item"
      >
        <ion-col>
          <div
            class="policy-meta-content"
          >
            <span
              class="subtitle policy-number"
              data-test="policy-number"
            >
               AZSS294857564 
            </span>
            <span
              class="subtitle"
            >
              <csaa-policy-subtitles>
                <span
                  class="subtitle"
                >
                  test
                </span>
              </csaa-policy-subtitles>
            </span>
          </div>
        </ion-col>
      </ion-row>
    </ion-grid>
    <ion-list>
      <ion-item
        class="ion-no-padding"
        data-test="button-pay-now"
        detail=""
      >
        <ion-label>
          Pay now
        </ion-label>
      </ion-item>
      <ion-item
        class="csaa-payment-autopay-item ion-no-padding"
        data-test="button-autopay"
        detail=""
      >
        <ion-label>
          <label>
            AutoPay
          </label>
          <small>
          </small>
        </ion-label>
        <span
          slot="end"
        >
           On
        </span>
      </ion-item>
    </ion-list>
  </div>
</csaa-make-payment-item>
`;

exports[`CsaaMakePaymentItemComponent should match snapshots when payment is overdue 1`] = `
<csaa-make-payment-item
  __ngContext__={[Function LRootView]}
  autoPayClick={[Function EventEmitter_]}
  customerSearch={[Function Object]}
  loading="false"
  payNowClick={[Function EventEmitter_]}
  payment={[Function Object]}
  paymentMethod={[Function Object]}
  paymentSucceeded="false"
  timezone="undefined"
>
  <div
    class="csaa-make-payment-item"
  >
    <ion-grid>
      <ion-row
        class="item"
      >
        <ion-col>
          <span
            class="payment-due overdue"
            data-test="payment-due"
          >
            Past
             Due August 19, 2040 
          </span>
        </ion-col>
        <ion-col
          size="auto"
        >
          <span
            class="payment-due overdue"
            data-test="payment-amount"
          >
            $148.10
          </span>
        </ion-col>
      </ion-row>
      <ion-row
        class="item"
      >
        <ion-col>
          <div
            class="policy-meta-content"
          >
            <span
              class="subtitle policy-number"
              data-test="policy-number"
            >
               AZSS294857564 
            </span>
            <span
              class="subtitle"
            >
              <csaa-policy-subtitles>
                <span
                  class="subtitle"
                >
                  test
                </span>
              </csaa-policy-subtitles>
            </span>
          </div>
        </ion-col>
      </ion-row>
    </ion-grid>
    <ion-list
      class="overdue"
    >
      <ion-item
        class="ion-no-padding"
        data-test="button-pay-now"
        detail=""
      >
        <ion-label>
          Pay now
        </ion-label>
      </ion-item>
      <ion-item
        class="csaa-payment-autopay-item ion-no-padding"
        data-test="button-autopay"
        detail=""
      >
        <ion-label>
          <label>
            AutoPay
          </label>
          <small>
          </small>
        </ion-label>
        <span
          slot="end"
        >
           On
        </span>
      </ion-item>
    </ion-list>
  </div>
</csaa-make-payment-item>
`;
